### PR TITLE
fix nonunique bug

### PR DESCRIPTION
**Pull request by bkamins**
fixes https://github.com/JuliaData/DataFrames.jl/issues/3392

reproducer:
```
julia> nonunique(DataFrame(a=PooledArray(Int[])))
ERROR: InexactError: trunc(Int64, NaN)
```

### CODE REVIEW

File: src/groupeddataframe/utils.jl [hunk 1]
```diff
@@ -337,7 +337,11 @@ function row_group_slots!(cols::NTuple{N, AbstractVector},
        nt = max(1, lg ÷ 100_000)
     end
     # if there are few rows per group limit the number of threads used
-    nt = clamp(round(Int, (lg / 4) / ngroups - 2), 1, nt)
+    if ngroups == 0
```
Comment (bkamins) [inline src/groupeddataframe/utils.jl:340]: theoretically we should never reach the case `ngroups == 0` here, as I added the `nrow` check in `nonunique`, but for future safety it is better to change this code also.

Comment (nalimilan) [reply src/groupeddataframe/utils.jl:340]: Maybe make it an `@assert` then?

Comment (bkamins) [reply src/groupeddataframe/utils.jl:340]: The point is that for `row_group_slots!` this is a valid input. It is just in the "outer" functions we usually special case this situation, but we do not guarantee it.

I have devised the test that triggers the error:
```
julia> groupby(DataFrame(x=categorical([missing])), :x, skipmissing=true)
ERROR: InexactError: trunc(Int64, Inf)
```
I have added the test so we will cover this line 😄.

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**